micro_http/src/server.rs

```diff
@@ -0,0 +1,752 @@
+            }
+            client_connection.enqueue_response(response.response);
+        }
+        Ok(())
+    }
+
+    /// Accepts a new incoming connection and adds it to the `epoll` notification structure.
+    fn handle_new_connection(&mut self) -> Result<()> {
+        if self.connections.len() == MAX_CONNECTIONS {
+            // If we want a replacement policy for connections
+            // this is where we will have it.
+            return Err(ServerError::ServerFull);
+        }
+
+        self.socket
+            .accept()
+            .map_err(ServerError::IOError)
+            .and_then(|(stream, _)| {
+                // `HttpConnection` is supposed to work with non-blocking streams.
+                stream
+                    .set_nonblocking(true)
+                    .map(|_| stream)
+                    .map_err(ServerError::IOError)
+            })
+            .and_then(|stream| {
+                // Add the stream to the `epoll` structure and listen for bytes to be read.
+                Self::epoll_add(self.epoll_fd, stream.as_raw_fd())?;
+                // Then add it to our open connections.
+                self.connections.insert(
+                    stream.as_raw_fd(),
+                    ClientConnection::new(HttpConnection::new(stream)),
+                );
+                Ok(())
+            })
+    }
+
+    /// Changes the event type for a connection to either listen for incoming bytes
+    /// or for when the stream is ready for writing.
+    fn epoll_mod(epoll_fd: RawFd, stream_fd: RawFd, evset: epoll::Events) -> Result<()> {
+        let event = epoll::Event::new(evset, stream_fd as u64);
+        epoll::ctl(
+            epoll_fd,
+            epoll::ControlOptions::EPOLL_CTL_MOD,
+            stream_fd,
+            event,
+        )
+        .map_err(ServerError::IOError)
+    }
+
+    /// Adds a stream to the `epoll` notification structure with the `EPOLLIN` event set.
+    fn epoll_add(epoll_fd: RawFd, stream_fd: RawFd) -> Result<()> {
+        epoll::ctl(
+            epoll_fd,
+            epoll::ControlOptions::EPOLL_CTL_ADD,
+            stream_fd,
+            epoll::Event::new(epoll::Events::EPOLLIN, stream_fd as u64),
+        )
+        .map_err(ServerError::IOError)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+    use std::os::unix::net::UnixStream;
+
+    use common::Body;
+    use std::io::Read;
+    use std::io::Write;
+
+    #[test]
+    fn test_wait_one_connection() {
+        let path_to_socket = "/tmp/test_socket_http_server1.sock";
+        fs::remove_file(path_to_socket).unwrap_or_default();
+
+        let mut server = HttpServer::new(path_to_socket.to_string()).unwrap();
+        server.start_server().unwrap();
+
+        // Test one incoming connection.
+        let mut socket = UnixStream::connect(path_to_socket).unwrap();
+        server.incoming().unwrap();
```

Comment:
You are not using the response of this call. What do you expect it to be? Check for empty vector instead.

Reply:
I have updated all such usages of that function in tests.